templates/common/_base/files/kubelet-auto-sizing.yaml

```diff
@@ -0,0 +1,99 @@
+mode: 0755
+path: "/usr/local/sbin/dynamic-system-reserved-calc.sh"
+contents:
+  inline: |
+    #!/bin/bash
+    set -e
+    NODE_SIZES_ENV=${NODE_SIZES_ENV:-/etc/node-sizing.env}
+    function dynamic_memory_sizing {
+        total_memory=$(free -g|awk '/^Mem:/{print $2}')
+        # total_memory=8 test the recommended values by modifying this value
+        recommended_systemreserved_memory=0
+        if (($total_memory <= 4)); then # 25% of the first 4GB of memory
```

Comment:
Logic for `dynamic_memory_sizing` func seems a bit unreadable. Like condition `if (($total_memory <= 4))` has been used twice. Also, multiple if condition will be called based on what memory size for e.g. 2 G memory. Is this logic coming already testing in upstream?

Reply:
Talked offline to Harshal and it seems this is the correct approach.